xonsh/tools.py

```diff
@@ -205,6 +216,28 @@ def __eq__(self, other):
         return all(map(operator.eq, self, other))
 
 
+class EnvPathDir(EnvPath):
```

Comment:
I think I'd prefer to see `BASH_COMPLETIONS` change. To my mind, a `PATH` is a collection of directories containing executables -- the `BASH_COMPLETIONS` isn't, necessarily, because of how the lazy-loading in the later versions work.